core/src/main/java/io/grpc/ClientInterceptors.java

```diff
@@ -114,71 +116,93 @@ private ProcessInterceptorChannel(Channel channel, Iterable<ClientInterceptor> i
 
   /**
    * A {@link Call} which forwards all of it's methods to another {@link Call}.
+   *
+   * @deprecated Use {@link SimpleForwardingCall}.
    */
-  public static class ForwardingCall<ReqT, RespT> extends Call<ReqT, RespT> {
-
-    private final Call<ReqT, RespT> delegate;
-
+  @Deprecated
+  public static class ForwardingCall<ReqT, RespT> extends SimpleForwardingCall<ReqT, RespT> {
     public ForwardingCall(Call<ReqT, RespT> delegate) {
-      this.delegate = delegate;
+      super(delegate);
     }
+  }
 
+  private static final Call<Object, Object> NOOP_CALL = new Call<Object, Object>() {
     @Override
-    public void start(Listener<RespT> responseListener, Metadata.Headers headers) {
-      this.delegate.start(responseListener, headers);
-    }
+    public void start(Listener<Object> responseListener, Metadata.Headers headers) { }
 
     @Override
-    public void request(int numMessages) {
-      this.delegate.request(numMessages);
-    }
+    public void request(int numMessages) { }
 
     @Override
-    public void cancel() {
-      this.delegate.cancel();
-    }
+    public void cancel() { }
 
     @Override
-    public void halfClose() {
-      this.delegate.halfClose();
-    }
+    public void halfClose() { }
 
     @Override
-    public void sendPayload(ReqT payload) {
-      this.delegate.sendPayload(payload);
-    }
-  }
+    public void sendPayload(Object payload) { }
+  };
 
   /**
-   * A {@link Call.Listener} which forwards all of its methods to another
-   * {@link Call.Listener}.
+   * A {@link io.grpc.ForwardingCall} that delivers exceptions from its start logic to the call listener.
+   *
+   * <p>{@link io.grpc.ForwardingCall#start()} should not throw any exception other than those
+   * caused by misuse, e.g., {@link IllegalStateException}.  {@code CheckedForwardingCall} provides
+   * {@code checkedStart()} in which throwing exceptions is allowed.
    */
-  public static class ForwardingListener<T> extends Call.Listener<T> {
-
-    private final Call.Listener<T> delegate;
-
-    public ForwardingListener(Call.Listener<T> delegate) {
+  public abstract static class CheckedForwardingCall<ReqT, RespT>
+      extends io.grpc.ForwardingCall<ReqT, RespT> {
+
+    private Call<ReqT, RespT> delegate;
+
+    /**
+     * Subclasses implement the start logic here that would normally belong to {@code start()}.
+     *
+     * <p>Implementation should call {@code this.delegate().start()} in the normal path. If the code
```

Comment:
We need to be stronger with the exception handling notices. Maybe more like:
```
<p>Implementation should call {@code this.delegate().start()} in the normal path. Exceptions may safely
be thrown prior to calling {@code this.delegate().start()}. Such exceptions will be handled by {@code
CheckedForwardingCall and be delivered to {@code responseListener}. Exceptions <em>must not</em>
be thrown after calling {@code this.delegate().start()}, as this can result in {@link Call.Listener#onClose}
being called multiple times.
```